code/modules/mob/mob_defines.dm

```diff
@@ -128,3 +128,13 @@
 
 	var/flavor_text = ""
 	var/flavor_text_2 = "" //version of the above that only lasts for the current round.
+
+	///////TYPING INDICATORS///////
+	/// Set to true if we want to show typing indicators.
+	var/typing_indicator_enabled = FALSE
+	/// Default icon_state of our typing indicator. Currently only supports paths (because anything else is, as of time of typing this, unnecesary.
+	var/typing_indicator_state = /obj/effect/overlay/typing_indicator
+	/// The timer that will remove our indicator for early aborts (like when an user finishes their message)
+	var/typing_indicator_timerid
+	/// Default typing indicator timeout
+	var/typing_indicator_timeout = 30 SECONDS
```

Comment:
That's too much time, 6-8 seconds would suffice, the chatbox is not really suitable for long messages. Also any reason this shouldn't be a define instead?

EDIT: oh wait, does this things actually check keybindings instead of the chatbox?

Reply:
yes